### PR TITLE
New module subcommand : mageck mle

### PR DESCRIPTION
**Pull request by LaurenceKuhl**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/mageck/mle/main.nf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        'https://depot.galaxyproject.org/singularity/mageck:0.5.9--py37h6bb024c_0':
+        'quay.io/biocontainers/mageck:0.5.9--py37h6bb024c_0' }"
+
+    input:
+    tuple val(meta), path(count_table)
+    path(design_matrix)
+
+    output:
+    tuple val(meta), path("*.gene_summary.txt"), emit: gene_summary
+    path "versions.yml"           , emit: versions
```
Comment (nvnieuwk) [inline modules/nf-core/mageck/mle/main.nf:16]:
```suggestion
    tuple val(meta), path("*.gene_summary.txt"), emit: gene_summary
    path "versions.yml"                        , emit: versions
```

File: modules/nf-core/mageck/mle/meta.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: "mageck_mle"
+description: maximum-likelihood analysis of gene essentialities computation
+keywords:
+  - sort
```
Comment (nvnieuwk) [inline modules/nf-core/mageck/mle/meta.yml:4]: Can you specify more keywords?

File: modules/nf-core/mageck/mle/meta.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+tools:
+  - "mageck":
+      description: "MAGeCK (Model-based Analysis of Genome-wide CRISPR-Cas9 Knockout), an algorithm to process, QC, analyze and visualize CRISPR screening data."
+      homepage: "https://sourceforge.net/p/mageck/wiki/Home/#mle"
+      documentation: "https://sourceforge.net/p/mageck/wiki/Home/"
+      tool_dev_url: "https://bitbucket.org/liulab/mageck/src"
+      doi: " https://doi.org/10.1186/s13059-015-0843-6"
```
Comment (nvnieuwk) [inline modules/nf-core/mageck/mle/meta.yml:11]:
```suggestion
      doi: "https://doi.org/10.1186/s13059-015-0843-6"
```

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

**Comment by nvnieuwk** [comment]
@nf-core-bot fix linting

**Comment by LaurenceKuhl** [comment]
@nvnieuwk  cool thanks for the bot command, didn't know that one!
I ran `nf-core modules create-test-yml -t mageck/mle` to create the tests but it didn't compute the checksum for my output file, would you know why that is ? mageck mle is a straightforward tool there shouldn't be a file difference

**Comment by nvnieuwk** [comment]
> @nvnieuwk cool thanks for the bot command, didn't know that one! I ran `nf-core modules create-test-yml -t mageck/mle` to create the tests but it didn't compute the checksum for my output file, would you know why that is ? mageck mle is a straightforward tool there shouldn't be a file difference

You're welcome :) it only works for linting errors caused by Prettier though. 
That's strange, I'll investigate

**Comment by nvnieuwk** [comment]
`nf-core modules create-test-yml mageck/mle` has now been running for about 20 minutes in a gitpod environment... Does it also take that long for you offline?

**Comment by nvnieuwk** [comment]
Strange after about 30 minutes the first test finished. The repeat test is now running

**Comment by LaurenceKuhl** [comment]
Ok no for me it ran and was way quicker, it just didn't compute the checksum in the `test.yml`

**Comment by nvnieuwk** [comment]
Allright, it finished. It seems the output file doesn't stay consistent when the second test ran. Lots of lines differ when I checked using `diff` of the two output files. Is this expected behaviour?

**Comment by nvnieuwk** [comment]
Can you also add an output channel for the `*.sgrna_summary.txt`?

**Comment by LaurenceKuhl** [comment]
Hi i tried locally and also got different files, i had a look at the article and source code and there seems to be a part with bayesian statistics, which would then make the results not totally the same every time (I think?), i tried just putting a keyword in the test file

File: modules/nf-core/mageck/mle/main.nf [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        'https://depot.galaxyproject.org/singularity/mageck:0.5.9--py37h6bb024c_0':
+        'quay.io/biocontainers/mageck:0.5.9--py37h6bb024c_0' }"
+
+    input:
+    tuple val(meta), path(count_table)
+    path(design_matrix)
+
+    output:
+    tuple val(meta), path("*.gene_summary.txt"), emit: gene_summary
+    path "versions.yml"            , emit: versions
```
Comment (nvnieuwk) [inline modules/nf-core/mageck/mle/main.nf:16]:
```suggestion
    tuple val(meta), path("*.gene_summary.txt") , emit: gene_summary
    tuple val(meta), path("*.sgrna_summary.txt"), emit: sgrna_summary
    path "versions.yml"                         , emit: versions
```

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

**Comment by nvnieuwk** [comment]
@nf-core-bot fix linting

File: tests/modules/nf-core/mageck/mle/test.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+- name: mageck mle test_mageck_mle
+  command: nextflow run ./tests/modules/nf-core/mageck/mle -entry test_mageck_mle -c ./tests/config/nextflow.config -c ./tests/modules/nf-core/mageck/mle/nextflow.config
+  tags:
+  - mageck
+  - mageck/mle
+  files:
+  - path: output/mageck/test.gene_summary.txt
+    contains: '['Gene']'
```
Comment (nvnieuwk) [inline tests/modules/nf-core/mageck/mle/test.yml:8]:
```suggestion
    contains: ['Gene']
```

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

**Comment by nvnieuwk** [comment]
@nf-core-bot fix linting

**Review by nvnieuwk — APPROVED** [review]
LGTM :tada:

**Pull request merged**